cmd/activator/request_log.go

```diff
@@ -29,11 +30,20 @@ import (
 
 func updateRequestLogFromConfigMap(logger *zap.SugaredLogger, h *pkghttp.RequestLogHandler) func(configMap *corev1.ConfigMap) {
 	return func(configMap *corev1.ConfigMap) {
-		newTemplate := configMap.Data["logging.request-log-template"]
-		if err := h.SetTemplate(newTemplate); err != nil {
-			logger.Errorw("Failed to update the request log template.", zap.Error(err), "template", newTemplate)
+		var newTemplate string // default is an empty template, disables logging
+		obsconfig, err := metrics.NewObservabilityConfigFromConfigMap(configMap)
+		if err != nil {
+			logger.Errorw("Failed to get observability configmap.", zap.Error(err), "configmap", configMap)
 		} else {
-			logger.Infow("Updated the request log template.", "template", newTemplate)
+			if obsconfig.EnableRequestLog {
```

Comment:
If `EnableRequestLog` is false, `newTemplate` will always be empty. I think this is the logic we will roll out in 0.18, but not 0.17. Is my understanding correct?

Reply:
@yanweiguo correct but right now since the logic in observability config sets everything as expected for 0.17 and 0.18, this does not change anything but I will remove the check here and pass the value directly to simplify things. Yes this part here is expected for 0.18 with a proper default.